### ISSUE TITLE
Dataset state space clarifications

### ISSUE THREAD

**Issue by Michaelszeng** [arth-shukla/mshab#57]
Hi, thanks great work and awesome benchmarking suite! I want to train diffusion policies to perform sequential tasks. I have some clarifying questions about the choice of state space in the included datasets.

From #18 , I've gathered that the state and action spaces of Fetch are the following:

```
agent
     - qpos (12-dim)
     - qvel (12-dim)
extra
     - tcp_pose_wrt_base (7-dim), tcp = "tool center point", i.e. tip of robot's end effector
     - obj_pose_wrt_base (7-dim), zero-masked if no object for task
     - goal_pos_wrt_base (3-dim), zero-masked if no goal for task
     - is_grasped (1-dim)
sensor_param
     - fetch_head
           - extrinsic_cv
           - cam2world_gl
           - intrinsic_cv
     - fetch_hand
           - similar to head
sensor_data
     - fetch_head
          - rgb (128x128x3, unit8)
          - depth (128x128x1, int16)
     - fetch_hand
           - similar to head


qpos and qvel are ordered as follows:

root_x_axis_joint (dummy joint, excluded from obs)
root_y_axis_joint (dummy joint, excluded from obs)
root_z_rotation_joint (dummy joint, excluded from obs)
torso_lift_joint
head_pan_joint
shoulder_pan_joint
head_tilt_joint
shoulder_lift_joint
upperarm_roll_joint
elbow_flex_joint
forearm_roll_joint
wrist_flex_joint
wrist_roll_joint
r_gripper_finger_joint
l_gripper_finger_joint
```

Additionally, I've found that only the depth images are contained in the dataset.

My questions are:
- Why did you choose not include 2D pose of the robot base in the dataset, and why _do_ you choose include information like `obj_pose_wrt_base`? Have you tried training policies with 2D pose of the robot base and/or without ground-truth information like `obj_pose_wrt_base` or `is_grasped`?
- Why do you not use RGB images and have you tried training policies using RGB (with or without depth)?
- Lastly, do you have datasets on sequential/long-horizon tasks? Or will we have to generate them this ourselves?

Thank you!

**Comment by arth-shukla** [comment]
Hi @Michaelszeng, thanks for your interest!

The dataset should contain both rgb and depth images -- is there a file you're checking which does not include RGB? Please let me know if so!

Regarding your questions:

1. Obs space

We use ground truth `obj_pose_wrt_base` and `is_grasped` to make the RL problem tractable. Furthermore, these have reasonable real-world equivalents: obj pose can be determined with pose estimation, and is_grasped by checking if the gripper is unable to close further around an object.

We didn't add obs like ground truth 2D position since they didn't provide much benefit compared to obj positions wrt base. For the manipulation tasks, where the robot spawns within 2m of the target obj, the 2d base position in the whole scene is not very useful. For cross-apartment navigation this may be more helpful, but here we chose to keep obs space consistent with the other tasks.

2. RGB

We trained RL on depth to save memory (SAC requires a large replay buffer, RGB-D took too much memory, and distances are harder to estimate with RGB alone). Depth also has a smaller sim2real gap than RGB, so the obs are somewhat more realistic. However, the dataset does also include RGB if you'd like to use that.

3. Long-horizon dataset

We haven't released a long-horizon dataset since the current baselines don't solve the long-horizon tasks well enough for good data coverage. This will require improved mobile manipulation methods. However, there is a long-horizon data gen script in case this data is needed

Let me know if you have more questions!

**Comment by Garsonjw** [comment]
Hi, I’d like to understand the purpose of these three virtual joints. When I use the 13-dimensional action and the 42-dimensional state to train other models, the performance becomes very poor. From the rollout videos, it seems that the base movement and the arm manipulation are not aligned with each other.

**Comment by Garsonjw** [comment]
In addition, I’d like to know whether the configuration for training the open action is exactly the same as the pick configuration provided in the repository.

**Comment by arth-shukla** [comment]
@Garsonjw Sorry for the late reply! It's been a busy time with NeurIPS and some deadlines.

Re: configs for training, each policy comes with a yml with the configs used to train that policy. Pick used SAC and Open used PPO. More details are available in A.4.3 of the paper.

Re: the virtual joints, these are used to model the base, but should not be included in obs (we remove from obs in [_get_obs_agent](https://github.com/arth-shukla/mshab/blob/07602db16f1885648e1f32b321ad12371c66286e/mshab/envs/sequential_task.py#L1456)).

Re: training performance, you can try using the configs provided with the checkpoints as a starting point, these include tuned hyperparameters. Are you using SAC/PPO to train, or a separate method?

**Comment by liujiahao2077** [comment]
What do the different dimensions in the action space represent? If I remember correctly, the actions in the dataset are incremental actions (except for the gripper). So why are two values very large when the robot is stationary?
<img width="917" height="87" alt="Image" src="https://github.com/user-attachments/assets/050e8709-d796-4f52-b994-4aaaf540f270" />

**Comment by arth-shukla** [comment]
@liujiahao2077 The value of each action dim can be referenced for each control mode in fetch.py here: https://github.com/haosulab/ManiSkill/blob/main/mani_skill/agents/robots/fetch/fetch.py#L259

Each action config dict is flattened to create the final action space.

As for large action outputs in the attached screenshot, for joint delta pos control, larger delta pos targets will change the PD controller's joint pos target by a larger amount. So, large action values result in joints rotating faster.
